pytest_inmanta/plugin.py

```diff
@@ -659,6 +662,12 @@ def __init__(
         self._handlers: typing.Set[ResourceHandler] = set()
         config.Config.load_config()
 
+    def get_compiler_executable(self) -> str:
+        if sys.platform != "win32":
+            return os.path.join(self._env_path, "bin", "python")
```

Comment:
I think on some systems `python` does not exist, only `python3`. On core we have `python_name: str = os.path.basename(sys.executable)`. Full snippet:
```
    @classmethod
    def get_python_path_for_env_path(cls, env_path: str) -> str:
        """
        For the given venv directory (`env_path`) return the path to the Python interpreter.
        """
        python_name: str = os.path.basename(sys.executable)
        return (
            os.path.join(env_path, "Scripts", python_name)
            if sys.platform == "win32"
            else os.path.join(env_path, "bin", python_name)
        )
```

Reply:
That as well. If we also use `os.path.basename(sys.executable)` here that should be fine. Or is that not what you mean?

Reply:
What I meant was this:
`os.path.basename(sys.executable)` returns `python3`
But in the compiler's venv there is no `python3` , only 'python' 
And so when we return `os.path.join(env_path, "bin", python_name)`, this path doesn't actually exist.

Reply:
I think venvs always have all levels of qualification so this wouldn't be an issue. I'm not completely sure of this but it is definitely true on my system:
```
sander@bedevere:~/downloads$ rm -rf env && python -m venv env && ls env/bin/python*
env/bin/python  env/bin/python3  env/bin/python3.10
```

Reply:
On a Rocky 8 host, which does not have `python`, only `python3`:
```
[rocky@ci-pytest-inmanta-lsm-iso6-dev ~]$ rm -rf env && python3 -m venv env && ls env/bin/python*
env/bin/python  env/bin/python3  env/bin/python3.9
```

Reply:
Ooh, right and `python3` and `python3.9` symlink to `python`

Reply:
```
sander@bedevere:~$ ll downloads/env/bin/python*
lrwxrwxrwx 1 sander sander 15 Apr 25 13:40 downloads/env/bin/python -> /usr/bin/python*
lrwxrwxrwx 1 sander sander  6 Apr 25 13:40 downloads/env/bin/python3 -> python*
lrwxrwxrwx 1 sander sander  6 Apr 25 13:40 downloads/env/bin/python3.10 -> python*
```
Not entirely, but semantically the same.